src/main/webapp/app/entities/course/course.component.ts

```diff
@@ -51,6 +56,35 @@ export class CourseComponent implements OnInit, OnDestroy {
         this.eventSubscriber = this.eventManager.subscribe('courseListModification', () => this.loadAll());
     }
 
+    /**
+     * Opens delete course dialog
+     * @param course that will be deleted
+     */
+    openDeleteCourseDialog(course: Course) {
+        if (!course) {
+            return;
+        }
+        const deleteDialogData: DeleteDialogData = {
+            entityTitle: course.title,
+            deleteQuestion: 'artemisApp.course.delete.question',
+            deleteConfirmationText: 'artemisApp.course.delete.typeNameToConfirm',
+        };
+        this.deleteDialogService.openDeleteDialog(deleteDialogData).then(
+            () => {
+                this.courseService.delete(course.id).subscribe(
+                    () => {
+                        this.eventManager.broadcast({
+                            name: 'courseListModification',
+                            content: 'Deleted an course',
+                        });
+                    },
+                    error => this.onError(error),
+                );
+            },
+            () => {},
```

Comment:
You can just remove the noop here, please also do this for the other places.

Reply:
Since I moved it to DeleteDialogButton, this is not relevant anymore